packages/html-reporter/src/reportView.tsx

```diff
@@ -42,51 +43,75 @@ const testCaseRoutePredicate = (params: URLSearchParams) => params.has('testId')
 export const ReportView: React.FC<{
   report: LoadedReport | undefined,
 }> = ({ report }) => {
-  const searchParams = new URLSearchParams(window.location.hash.slice(1));
+  const searchParams = useSearchParams();
   const [expandedFiles, setExpandedFiles] = React.useState<Map<string, boolean>>(new Map());
   const [filterText, setFilterText] = React.useState(searchParams.get('q') || '');
 
+  const reportData = report?.json();
   const filter = React.useMemo(() => Filter.parse(filterText), [filterText]);
-  const filteredStats = React.useMemo(() => computeStats(report?.json().files || [], filter), [report, filter]);
+
+  const filteredFiles = React.useMemo(() => {
```

Comment:
Looks like it would read much better if you computed `filteredFiles` in `computeStats`.

Reply:
Sorry, I'm not sure what you have in mind. Could you clarify?

`filteredFiles` is also passed to child components so why would I compute that in `computeStats`?

Reply:
I meant to save a traversal - you now traverse tests twice and filtered files + stats could be combined into a single processed file set struct. It is fine to keep it your way though.